packages/manager/modules/dbaas-logs/src/logs/detail/streams/subscriptions/streams-subscriptions.controller.js

```diff
@@ -0,0 +1,126 @@
+    this.ouiDatagridService = ouiDatagridService;
+    this.CucCloudMessage = CucCloudMessage;
+    this.CucControllerHelper = CucControllerHelper;
+    this.LogsHelperService = LogsHelperService;
+    this.LogsStreamsService = LogsStreamsService;
+    this.LogsStreamsSubscriptionsService = LogsStreamsSubscriptionsService;
+
+    this.serviceName = this.$stateParams.serviceName;
+    this.streamId = this.$stateParams.streamId;
+    this.initLoaders();
+  }
+
+  $onInit() {
+    this.stream.load();
+  }
+
+  /**
+   * Load the current stream
+   */
+  initLoaders() {
+    this.stream = this.CucControllerHelper.request.getHashLoader({
+      loaderFunction: () =>
+        this.LogsStreamsService.getStream(this.serviceName, this.streamId),
+    });
+  }
+
+  /**
+   * Retrieve subscription list, according to pagination/sort/filter
+   *
+   * @param offset int element offset to retrieve results from
+   * @param pageSize int Number of results to retrieve
+   * @param sort Object Sort object from ovh-ui datagrid
+   * @param criteria Object Criteria object from ovh-ui datagrid
+   * @return {*|Promise<any>}
+   */
+  loadSubscriptions({ offset, pageSize = 1, sort, criteria }) {
+    const filters = criteria.map((c) => {
+      const name = c.property || 'resource.name';
+      return datagridToIcebergFilter(name, c.operator, c.value);
+    });
+    const pageOffset = Math.ceil(offset / pageSize);
+    return this.LogsStreamsSubscriptionsService.getPaginatedStreamSubscriptions(
+      this.serviceName,
+      this.streamId,
+      pageOffset,
+      pageSize,
+      { name: sort.property, dir: sort.dir === -1 ? 'DESC' : 'ASC' },
+      filters,
+    );
+  }
+
+  /**
+   * Display a modal to confirm subscription deletion
+   *
+   * @param subscription Object Subscription object from API
+   * @return {*|Promise<any>}
+   */
+  showSubscriptionDeleteConfirm(subscription) {
+    this.CucCloudMessage.flushChildMessage();
+    return this.CucControllerHelper.modal
+      .showDeleteModal({
+        titleText: this.$translate.instant(
+          'streams_subscriptions_delete_modal_title',
+        ),
+        textHtml: this.$translate.instant(
+          'streams_subscriptions_delete_modal_content',
+          {
+            resourceName: `<strong>${subscription.resource.name}</strong>`,
+          },
+        ),
+      })
+      .then(() => this.removeSubscription(subscription));
+  }
+
+  /**
+   * Delete a subscription on API
+   * Update datagrid accordingly
+   *
+   * @param subscription Object Subscription object from API
+   */
+  removeSubscription(subscription) {
+    this.CucCloudMessage.flushChildMessage();
+    this.deleteSubscriptionLoading = true;
+    this.LogsStreamsSubscriptionsService.deleteSubscription(
+      this.serviceName,
+      this.streamId,
+      subscription,
+    )
+      .then((operation) =>
+        this.LogsHelperService.handleOperation(
+          this.serviceName,
+          operation.data || operation,
```

Comment:
```suggestion
          operation.data,
```
What's the case where response result is not embedded in `data` property ?

Reply:
Done. As you said the `data` property should always be there, but this check is used everywhere in this module:

```
› grep -R "operation.data || operation" packages/manager/modules/dbaas-logs 
packages/manager/modules/dbaas-logs/src/logs/detail/encryption-keys/home/logs-encryption-keys-home.controller.js:          operation.data || operation,
packages/manager/modules/dbaas-logs/src/logs/detail/encryption-keys/add/logs-encryption-keys-add.controller.js:          operation.data || operation,
packages/manager/modules/dbaas-logs/src/logs/detail/index/logs-index.service.js:          operation.data || operation,
packages/manager/modules/dbaas-logs/src/logs/detail/index/logs-index.service.js:          operation.data || operation,
packages/manager/modules/dbaas-logs/src/logs/detail/index/logs-index.service.js:          operation.data || operation,
...
```